dali/operators/video/frames_decoder_base.cc

```diff
@@ -603,20 +603,21 @@ void FramesDecoderBase::DecodeFramesImpl(uint8_t *data,
   DALI_ENFORCE(constant_frame != nullptr || boundary_type != boundary::BoundaryType::CONSTANT,
                make_string("Constant frame must be provided if boundary type is CONSTANT"));
 
-  int last_out_frame_idx = -1;
+  uint8_t* last_out_frame_start = nullptr;
   for (auto &[frame_id, i] : frame_ids) {
+    auto out_frame_start = data + ptrdiff_t(i) * FrameSize();
```

Comment:
```suggestion
    uint8_t *out_frame_start = data + ptrdiff_t(i) * FrameSize();
```
...since we assign it to explicitly typed last_out_frame_start...